aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -361,24 +364,24 @@ void upsample_avx_bilinear_uint8(
   // Minor optimization: we can avoid allocating an extra buffer if we're performing
   // horizontal-only or vertical-only interpolation, and if the tensor doesn't
   // need unpacking
-  if (need_horizontal && !(needs_unpacking && !need_vertical)) {
-    auto c = (needs_unpacking) ? num_channels : 4;
+  if (need_horizontal && !(skip_packing && !need_vertical)) {
```

Comment:
this is quite hard to read IMHO. Is this easier?

```suggestion
  if (need_horizontal && (!skip_packing || need_vertical)) {
```